lib/VeresOneDidDoc.js

```diff
@@ -75,35 +75,36 @@ class VeresOneDidDoc {
    *
    * @returns {Promise}
    */
-  async init({mode, passphrase, didType, keyType = constants.DEFAULT_KEY_TYPE}) {
+  async init({
+    mode, passphrase, didType, keyType = constants.DEFAULT_KEY_TYPE
+  }) {
     const keyOptions = {type: keyType, passphrase};
 
     // Generate a capabilityInvocation key, to base the DID URI on
     const invokeKey = await LDKeyPair.generate(keyOptions);
     const did = this.generateId({keyPair: invokeKey, didType, mode});
     this.doc.id = did;
 
+    // assign the DID as the controller for the keys
+    invokeKey.controller = did;
+    keyOptions.controller = did;
+
     // Generate an authentication key pair and suite
     const authKey = await LDKeyPair.generate(keyOptions);
     authKey.id = this.generateKeyId({did, keyPair: authKey});
-    this.doc[constants.SUITES.authentication] = [
-      this.generateKeyObject({key: authKey})
-    ];
+    this.doc[constants.SUITES.authentication] = [authKey.publicNode()];
```

Comment:
Would be good to rename `constants.SUITES` here to `constants.PROOF_PURPOSES` or similar.

Reply:
@dlongley if you're saying the suites terminology needs to go, then we need to change APIs like `suiteKeyNode` here as well? https://github.com/veres-one/did-veres-one/blob/fixFingerprint/lib/VeresOneDidDoc.js#L160

Reply:
Yes, it should all change.